posts/app.py

```diff
@@ -66,13 +66,26 @@ def get_all_posts():
     return serialize_posts_to_json(post_list)
 
 
-@app.route('/v1/<post_id>', methods=['GET'])
-def get_post_by_id(post_id):
-    """Get the post with the specified ID."""
-    # serialize otherwise nonserializable ObjectIDs
-    post_list = find_posts_in_db(
-        app.config["COLLECTION"], post_id=ObjectId(post_id))
-    return serialize_posts_to_json(post_list)
+@app.route('/v1/<post_id>', methods=['GET', 'DELETE'])
+def get_delete_post_by_id(post_id):
```

Comment:
It seems risky to have Get and Delete endpoints using the same method. It seems too easy to introduce a bug where you delete every post you try to get.  I would separate out into individual get and delete methods, at least for the outer methods. They could share an inner helper method if the code is mostly the same.

Reply:
Done https://github.com/knative-portability/large-events/pull/52/commits/951624c1dc9ebdd0faf98cc852d060df85b8abf3. I split the delete endpoint to `DELETE /v1/<post_id>/delete`.